src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5747,7 +5842,23 @@ private BoundExpression BindTypeParameterCreationExpression(ObjectCreationExpres
                 }
             }
 
-            diagnostics.Add(errorLocation, useSiteInfo);
+            if (suppressUnsupportedRequiredMembersError && useSiteInfo.AccumulatesDiagnostics && useSiteInfo.Diagnostics is { Count: not 0 })
```

Comment:
I didn't understand the benefit of this suppression. If a constructor gets picked and it has a problem, why not report it?

Reply:
Because then you'll get the error, even if you decorate the constructor with `SetsRequiredMembers` (which will be supported in a later pr.

Reply:
Thanks. Consider leaving a comment to that effect